projects/fine_tuning/toolbox/fine_tuning_run_fine_tuning_job/tasks/main.yml

```diff
@@ -362,3 +362,37 @@ elif fine_tuning_run_fine_tuning_job_workload == 'ilab' -%}
          > {{ artifact_extra_logs_dir }}/artifacts/pod_{{ item | replace('pod/', '') }}.log
     failed_when: false
     loop: "{{ job_pod_name_cmd.stdout_lines }}"
+
+  - name: Retrieve Pod files to artifacts directory
+    when: fine_tuning_run_fine_tuning_job_retrieve_files | bool
+    block:
+      - name: Create the Pod artifacts directory
+        file:
+          path: "{{ artifact_extra_logs_dir }}/pod-artifacts/"
+          state: directory
+          mode: '0755'
+
+      - name: Launch fine-tuning-file-retrieval Pod from template
+        command:
+          oc create -f <(echo "{{ fine_tuning_job_retrieval_pod_template | to_json }}")
+
+      - name: Wait for fine-tuning-file-retrieval Pod to be Running
+        command:
+          set -o pipefail;
+
+          oc get pod --no-headers
+             fine-tuning-file-retrieval
+             -n {{ fine_tuning_run_fine_tuning_job_namespace }}
+             | awk '{print $3}'
+        register: retrieve_pod_state_cmd
+        until: '"Running" not in retrieve_pod_state_cmd.stdout_lines'
+        retries: 18
+        delay: 10
+
+      - name: Copy files to the Pod artifacts directory
+        command:
+          oc cp fine-tuning-file-retrieval:/mnt/storage/output/ "{{ artifact_extra_logs_dir }}/pod-artifacts/"
+
+      - name: Delete fine-tuning-file-retrieval Pod
+        command:
+          oc delete pod fine-tuning-file-retrieval
```

Comment:
did you notice that this part of the `always` block, which launched the fine-tuning job?
this is correct, it means that even if the pod failed, the files will be exported

that makes me realize that you write it this way:
```
block:
  Wait for fine-tuning-file-retrieval Pod to be Running
  Copy files to the Pod artifacts directory
always:
  Delete fine-tuning-file-retrieval Pod
```
so that the Pod is always deleted